src/component/mxgraph/shape/event-shapes.ts

```diff
@@ -24,44 +24,63 @@ const mxUtils: typeof mxgraph.mxUtils = MxGraphFactoryService.getMxGraphProperty
 const mxConstants: typeof mxgraph.mxConstants = MxGraphFactoryService.getMxGraphProperty('mxConstants');
 
 abstract class EventShape extends mxEllipse {
+  // when all/more event types will be supported, we could move to a Record/MappedType
+  private iconPainters: Map<ShapeBpmnEventKind, (c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number) => void> = new Map([
+    [ShapeBpmnEventKind.MESSAGE, (c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number) => this.paintMessageIcon(c, x, y, w, h)],
+    [ShapeBpmnEventKind.TERMINATE, (c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number) => this.paintTerminateIcon(c, x, y, w, h)],
+  ]);
+  protected isUsingThrowIcons = false;
+
   protected constructor(bounds: mxgraph.mxRectangle, fill: string, stroke: string, strokewidth: number) {
     super(bounds, fill, stroke, strokewidth);
   }
 
   public paintVertexShape(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
+    this.markNonFullyRenderedEvents(c);
+    this.paintOuterShape(c, x, y, w, h);
+    this.paintInnerShape(c, x, y, w, h);
+  }
+
+  // This will be removed when managing the render of all events
+  private markNonFullyRenderedEvents(c: mxgraph.mxXmlCanvas2D): void {
     const eventKind = this.getBpmnEventKind();
-    // will be removed when managing the timer rendering
     if (eventKind == ShapeBpmnEventKind.TIMER) {
       c.setFillColor('green');
       c.setFillAlpha(0.3);
+    } // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    else if (eventKind == ShapeBpmnEventKind.MESSAGE && (this instanceof StartEventShape || this instanceof EndEventShape)) {
+      c.setFillColor('yellow');
+      c.setFillAlpha(0.3);
     }
-
-    this.paintOuterShape(c, x, y, w, h);
-    this.paintInnerShape(c, x, y, w, h);
   }
 
   protected paintOuterShape(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
     super.paintVertexShape(c, x, y, w, h);
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   protected paintInnerShape(c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number): void {
-    // do nothing by default
+    const iconPainter =
+      this.iconPainters.get(this.getBpmnEventKind()) || ((c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number) => this.paintEmptyIcon(c, x, y, w, h));
+    iconPainter(c, x, y, w, h);
```

Comment:
```suggestion
    const paintIcon =
      this.iconPainters.get(this.getBpmnEventKind()) || ((c: mxgraph.mxXmlCanvas2D, x: number, y: number, w: number, h: number) => this.paintEmptyIcon(c, x, y, w, h));
    paintIcon(c, x, y, w, h);
```

Reply:
why not :smile_cat: 
the proposed name is close to a method/function name, so it is maybe clearer for the reader of the code.